language-plutus-core/plutus-ir/Language/PlutusIR/Compiler/Datatype.hs

```diff
@@ -293,7 +317,9 @@ mkConstructor dty d@(Datatype _ _ tvs _ constrs) index = withEnclosing (Datatype
     argsAndTypes <- do
           -- these types appear *outside* the scope of the abstraction for the datatype, so we need to use the concrete datatype here
           -- see note [Abstract data types]
-        let argTypes = unveilDatatype (getType dty) d <$> constructorArgTypes constr
+        -- TODO: datacons' type normalization should only happen in one place, too much duplication otherwise
+        normConstr <- Norm.runNormalizeTypeM $ Norm.normalizeTypeM $ varDeclType constr
```

Comment:
I suggest just removing this check for now and leaving a fixme like the others. It's not a problem in practice, we might as well fix it along with the others. Plus, then we can check that we *don't* get script hash changes without this, which I think we shouldn't.